packages/wasm/crate/src/planner.rs

```diff
@@ -148,8 +148,23 @@ pub async fn plan_transaction(
 ) -> WasmResult<JsValue> {
     utils::set_panic_hook();
 
-    let request = TransactionPlannerRequest::decode(request)?;
+    let tx_planner_req = TransactionPlannerRequest::decode(request)?;
+    let fvk: FullViewingKey = FullViewingKey::decode(full_viewing_key)?;
+    let fee_asset_id = Id::decode(gas_fee_token)?;
+    let constants: DbConstants = serde_wasm_bindgen::from_value(idb_constants)?;
+    let storage = init_idb_storage(constants).await?;
+
+    let plan = plan_transaction_inner(storage, tx_planner_req, fvk, fee_asset_id).await?;
+
+    Ok(serde_wasm_bindgen::to_value(&plan)?)
+}
```

Comment:
Created `plan_transaction_inner()` in order to separate the serde serialization from the core logic of the function (that gets tested in the test suite).